src/libsystemd/sd-journal/catalog.c

```diff
@@ -171,6 +172,7 @@ static int finish_item(
                 if (ordered_hashmap_update(h, i, combined) < 0)
```

Comment:
Not your fault, but `ordered_hashmap_update()` does not trigger OOM. Could you fix it when you touch the code?